### DISCUSSION TITLE
Single report for cypress ci tests running in parallel

### DISCUSSION THREAD

**Discussion by Gayathri-Kesavakannan** [cypress-io/cypress#24537] [Questions and Help]
Hello 
I have E2E Cypress tests running in ci pipeline with parallelisation (3tags). Is it possible to generate/email only 1 report for all 3 tags 

Thanks

**Comment by portyu9** [comment]
Hi @Gayathri-Kesavakannan 

Yes. The usual pattern is to generate a report on every parallel worker, collect those files as CI artifacts, and merge them in a final job after all workers complete.

For example, with JUnit:

```js
reporter: 'junit',
reporterOptions: {
  mochaFile: 'results/cypress-[hash].xml',
}
```

The `[hash]` is important because Cypress processes specs separately and otherwise report files can overwrite each other.

Then:

1. run all parallel Cypress jobs;
2. upload each job's `results/` directory as a CI artifact;
3. download all report artifacts in a final job;
4. merge the XML/JSON reports using the merger appropriate to your reporter;
5. email/publish that single merged report.

Current Cypress documentation specifically mentions `junit-report-merger` for JUnit and `mochawesome-merge` for Mochawesome.

If Cypress Cloud itself is sufficient as the report, all parallel machines can instead be associated with the same recorded Cypress run, which already gives you one aggregated run view.

Please mark this answer as accepted if it helped, thank you.
